### PR TITLE
Add SQL Server to APM-DBM docs and clarify the modes

### PR DESCRIPTION
**Pull request by justiniso**
### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Review by alexandre-normand — APPROVED** [review]
Yes! 🚀

**Review by alexandre-normand — APPROVED** [review]
💯

File: content/en/database_monitoring/guide/connect_dbm_and_apm.md [hunk 1]
```diff
@@ -272,7 +281,10 @@ Follow the [.NET Framework tracing instructions][1] or the [.NET Core tracing in
 Ensure that you are using a supported client library. For example, `Npgsql`.
 
 Enable the database monitoring propagation feature by setting the following environment variable:
-   - `DD_DBM_PROPAGATION_MODE=full`
+   - For Postgres & MySQL: `DD_DBM_PROPAGATION_MODE=full`
```
Comment (drichards-87) [inline content/en/database_monitoring/guide/connect_dbm_and_apm.md:284]:
```suggestion
   - For Postgres and MySQL: `DD_DBM_PROPAGATION_MODE=full`
```

File: content/en/database_monitoring/guide/connect_dbm_and_apm.md [hunk 2]
```diff
@@ -16,43 +16,52 @@ This guide assumes that you have configured [Database Monitoring][1] and are usi
...
-
+APM tracer integrations support a *Propagation Mode*, which controls the amount of information passed from applications to the database.
+
+- `full` mode sends full trace information to the database, allowing you to investigate individual traces within DBM. This is the recommended solution for most integrations.
+- `service` mode sends the service name, allowing you to understand which services are the contributors to database load. This is the only supported mode for SQL Server applications.
+- `none` mode disables propagation and does not send any information from applications.
+
+| DD_DBM_PROPAGATION_MODE  | Postgres  |   MySQL   |  SQL Server  |
+|:-------------------------|:---------:|:---------:|:------------:|
+| `full`                   | {{< X >}} | {{< X >}} | unsupported  |
```
Comment (drichards-87) [inline content/en/database_monitoring/guide/connect_dbm_and_apm.md:36]:
```suggestion
| `full`                   | {{< X >}} | {{< X >}} |          |
```

**Review by drichards-87 — CHANGES_REQUESTED** [review]

**Review by drichards-87 — APPROVED** [review]

**Pull request merged**